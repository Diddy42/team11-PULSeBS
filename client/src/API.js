async function userLogin(username, password) {
    return new Promise((resolve, reject) => {
        fetch('http://localhost:80/server/api/login.php', {     //Da verificare
            method: 'POST',
            headers: {
                //'Content-Type': 'application/json',
                'Content-Type' : 'application/x-www-form-urlencoded; charset=UTF-8',
            },
            body: JSON.stringify({username: username, password: password}),
        }).then((response) => {
            response.json().then((obj) => { 
                console.log(obj);
                if(obj === 0 ) {        //Da modificare in base a quanto ritorna il server. Sarebbe utile che ritorni l'id o anche nome e cognome la visualizzare, e il ruolo
                    reject(obj);
                } else {
                    resolve(obj);    
                }
            })
            .catch((err) => { reject({ errors: [{ param: "Application", msg: "Cannot parse server response" }] }) });
        }).catch((err) => { reject({ errors: [{ param: "Server", msg: "Cannot communicate" }] }) });
    });
}

async function logout(){
	return new Promise((resolve, reject) => {
        fetch('http://localhost:80/server/api/logout.php')      //Da verificare
        .then((response) => {
			if(response.ok){
				resolve();
			}
			else{
				reject();
			}
       })
       .catch((err) => { 
      		 reject(err); 
       });
    });
}


async function getStudentLectures(studentId) {
    const url = "test"                  //Da definire
    const response = await fetch(url);
    const lectures = await response.json();
    if(response.ok) {
        return lectures;
    } else {
        let err = {status: response.status, errorObj: lectures};
        throw err; 
    }
}

async function getBooking(lectureId) {
    const url = "test"                  //Da definire
    const response = await fetch(url);
    const booking = await response.json();
    if(response.ok) {
        return booking;
    } else {
        let err = {status: response.status, errorObj: booking};
        throw err; 
    }
}

async function bookASeat(lectureId, studentId) {
    const url = "test"                  //Da definire
    const response = await fetch(url, {
        method: 'POST',
        headers: {
            //'Content-Type': 'application/json',
            'Content-Type' : 'application/x-www-form-urlencoded; charset=UTF-8',
        },
        body: JSON.stringify({lectureId: lectureId, studentId: studentId}),
    });
    
    try{
        const resJ = await response.json();
        if(response.ok) {
		    return resJ;
		} else {
		    let err = {status: response.status, errorObj: resJ};
		    throw err; 
		}
    }
    catch(e){
        console.log("Error in booking a seat: " + e);
        return 0;   //meaning: no new customer
    }
}

const API = {
    userLogin, logout, getStudentLectures, getBooking, bookASeat
};
export default API;
