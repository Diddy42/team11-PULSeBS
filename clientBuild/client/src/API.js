async function userLogin(username, password) {
    return new Promise((resolve, reject) => {
        fetch('http://localhost/server/src/api/Login.php', {
            method: 'POST',
            headers: {
                //'Content-Type': 'application/json',
                'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
            },
            body: JSON.stringify({ username: username, password: password }),
        }).then((response) => {
            response.json().then((obj) => {
                /*
                if(obj === 0 ) {        //Da modificare in base a quanto ritorna il server. Sarebbe utile che ritorni l'id o anche nome e cognome la visualizzare, e il ruolo
                    reject(obj);
                } else {
                    resolve(obj);    
                }
                */
                resolve(obj);
            })
                .catch((err) => { reject({ errors: [{ param: "Application", msg: "Cannot parse server response" }] }) });
        }).catch((err) => { reject({ errors: [{ param: "Server", msg: "Cannot communicate" }] }) });
    });
}

async function logout() {
    return new Promise((resolve, reject) => {
        fetch('http://localhost:80/server/api/logout.php')      //Da verificare
            .then((response) => {
                if (response.ok) {
                    resolve(0);
                }
                else {
                    reject();
                }
            })
            .catch((err) => {
                reject(err);
            });
    });
}

//Ritorna le lezioni prenotabili dallo studente
async function getBookableStudentLectures(studentId) {
    const url = "http://localhost/server/bookableLessons/" + studentId;
    const response = await fetch(url);
    const lectures = await response.json();
    if (response.ok) {
        if (lectures === 0) {
            return [];
        }
        return lectures;
    } else {
        let err = { status: response.status, errorObj: lectures };
        throw err;
    }
}

//Ritorna le prenotazioni effettuate dagli studenti per la lezione indicata (teacher side)
async function getBooking(lectureId) {
    const url = "http://localhost/server/bookedStudentsForLecture/" + lectureId;
    const response = await fetch(url);
    const booking = await response.json();
    if (response.ok) {
        if (booking === 0) {
            return [];
        }
        return booking;
    } else {
        let err = { status: response.status, errorObj: booking };
        throw err;
    }
}

// API to retrieve all the bookings of a student
async function getStudentBookings(studentId) {
    const url = "http://localhost/server/studentBookings/" + studentId;
    const response = await fetch(url);
    const booking = await response.json();
    if (response.ok) {
        if (booking === 0) {
            return [];
        }
        return booking;
    } else {
        let err = { status: response.status, errorObj: booking };
        throw err;
    }
}

//API per prenotare un posto a lezione
async function bookASeat(lectureId, studentId, date) {
    const url = "http://localhost/server/insertLecture"
    const response = await fetch(url, {
        method: 'POST',
        headers: {
            //'Content-Type': 'application/json',
            'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
        },
        body: JSON.stringify({
            idLesson: lectureId,
            idUser: studentId,
            date: date
        }),
    });

    try {
        const resJ = await response.json();
        if (response.ok) {
            return resJ;
        } else {
            let err = { status: response.status, errorObj: resJ };
            throw err;
        }
    }
    catch (e) {
        console.log("Error in booking a seat: " + e);
        return 0;   //meaning: no new customer
    }
}

// API to retrieve all the lectures of a teacher
async function getTeacherLectures(teacherId) {
    const url = "http://localhost/server/scheduledLecturesForTeacher/" + teacherId
    const response = await fetch(url);
    const lectures = await response.json();
    if (response.ok) {
        if (lectures === 0) {
            return [];
        }
        return lectures;
    } else {
        let err = { status: response.status, errorObj: lectures };
        throw err;
    }
}

async function deleteBooking(bookingId) {
    const url = "http://localhost/server/updateBooking/" + bookingId;
    return new Promise((resolve, reject) => {
        fetch(url, {
            method: 'PUT',
            headers: {
                //'Content-Type': 'application/json',
                'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
            },
            body: JSON.stringify({
            }),
        }).then((response) => {
            if (response.ok) {
                resolve(0);
            } else {
                reject({ errore: "Error in deleting a booking" });
            }
        }).catch((err) => { reject({ errors: [{ param: "Server", msg: "Cannot communicate" }] }) });
    });
}

async function deleteLecture(lectureId) {
    const url = "http://localhost/server/deleteLecture/" + lectureId;
    const response = await fetch(url, {
        method: 'PUT',
        headers: {
            //'Content-Type': 'application/json',
            'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
        },
        body: JSON.stringify({
        }),
    });
    const result = await response.json();
    if (response.ok) {
        if (result === 1) {
            return 1;
        } else {
            console.log("Too late to delete the lecture");
            return -1;
        }
    } else {
        let err = { status: response.status, errorObj: result };
        throw err;
    }
}

async function changeToOnline(lectureId) {
    const url = "http://localhost/server/changeToOnline/" + lectureId;
    const response = await fetch(url, {
        method: 'PUT',
        headers: {
            //'Content-Type': 'application/json',
            'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
        },
        body: JSON.stringify({
        }),
    });
    const result = await response.json();
    if (response.ok) {
        if (result === 1) {
            return 1;
        } else {
            console.log("Too late to change the lecture");
            return -1;
        }
    } else {
        let err = { status: response.status, errorObj: result };
        throw err;
    }
}

async function getAllCourses() {
    const url = "http://localhost/server/courses";

    const response = await fetch(url);
    const courses = await response.json();
    if (response.ok) {
        if (courses === 0) {
            return [];
        }
        return courses;
    } else {
        let err = { status: response.status, errorObj: courses };
        throw err;
    }
}

const ALL_COURSES_FILTER = "L.idCourse";
async function getBookingStatisticsByMonth(courseIds) {
    const url = `http://localhost/server/bookingStatistics/?filterTime=year,monthOfYear&filterCourse=${courseIds.toString()}`;

    const response = await fetch(url);
    const stats = await response.json();
    if (response.ok) {
        if (stats === 0) {
            return [];
        }
        return stats;
    } else {
        let err = { status: response.status, errorObj: stats };
        throw err;
    }
} 
async function getBookingStatisticsByWeek(courseIds) {
    const url = `http://localhost/server/bookingStatistics/?filterTime=year_month_week&filterCourse=${courseIds.toString()}`;

    const response = await fetch(url);
    const stats = await response.json();
    if (response.ok) {
        if (stats === 0) {
            return [];
        }
        return stats;
    } else {
        let err = { status: response.status, errorObj: stats };
        throw err;
    }
}
async function getBookingStatisticsByLesson(courseIds) {
    const url = `http://localhost/server/bookingStatistics/?filterTime=L.idLesson&filterCourse=${courseIds.toString()}`;

    const response = await fetch(url);
    const stats = await response.json();
    if (response.ok) {
        if (stats === 0) {
            return [];
        }
        return stats;
    } else {
        let err = { status: response.status, errorObj: stats };
        throw err;
    }
}

async function getCoursesOfTeacher(teacherId) {
    const url = "http://localhost/server/teacherCourses/" + teacherId;

    const response = await fetch(url);
    const courses = await response.json();
    if (response.ok) {
        if (courses === 0) {
            return [];
        }
        return courses;
    } else {
        let err = { status: response.status, errorObj: courses };
        throw err;
    }
}

const API = {
    userLogin, logout, getBookableStudentLectures, getBooking, getStudentBookings, bookASeat, deleteBooking, getTeacherLectures, deleteLecture, changeToOnline, getAllCourses, 
    getBookingStatisticsByMonth, getBookingStatisticsByWeek, getBookingStatisticsByLesson, ALL_COURSES_FILTER, getCoursesOfTeacher
};
export default API;
