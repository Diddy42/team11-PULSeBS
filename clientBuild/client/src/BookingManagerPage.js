import React from 'react';
import ApexChart from "react-apexcharts";
import { Container, ButtonGroup, ToggleButton } from 'react-bootstrap';
import Select from 'react-select';
import API from './API';

class BookingManagerPage extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            currentBookingFilter: 1, // Month = 1
            currentCancellationFilter: 4, // Month = 4
            selectedCourse: API.ALL_COURSES_FILTER
        };
    }

    componentDidMount = () => {
        API.getAllCourses().then((courses) => {
            this.setState({
                courses: courses
            });
        });

        this.setState({
            bookingSeries: undefined,
            cancellationSeries: undefined,
        });

        API.getBookingStatisticsByMonth(this.state.selectedCourse).then((stats) => {
            let data = []; let labels = [];
            stats.forEach(stat => {
                data.push(stat.average);
                labels.push(`${stat.monthOfYear}-${stat.year}`);
            });
            this.updateBookingGraphData(
                'Average number of booking per month',
                data,
                data, // tmp (attendance)
                labels
            )
        });

        this.updateCancellationGraphData(
            'Average number of cancellations per month',
            [40.5, 42.8, 45.7, 60.1, 50.2, 40.3, 30.2],
            ['Oct 2020', 'Nov 2020', 'Dec 2020', 'Jan 2021', 'Feb 2021', 'Mar 2021', 'Apr 2021']
        );
    }

    render() {
        return <>
            <Container fluid>
                <br />
                <h1 className="text-center">Historical data</h1>

                <Select
                    className="col-4 mx-auto my-4"
                    options={this.state.courses}
                    placeholder="Select a course..."
                    isClearable={true}
                    styles={{
                        // Fixes the overlapping problem of the component
                        menu: provided => ({ ...provided, zIndex: 9999 })
                    }}
                    getOptionLabel={(course) => course.courseName}
                    getOptionValue={(course) => course.idCourse}
                    onChange={selection => this.handleSelectionChange(selection)}
                />

                {
                    this.state.selectedCourse != API.ALL_COURSES_FILTER ?
                        <>
                            <div className='col-8 mx-auto my-5 shadow-lg p-3 bg-white'>
                                {this.state.bookingSeries != undefined &&
                                    <ApexChart options={this.state.bookingOptions} series={this.state.bookingSeries} type="area" height={350} />}
                                <div className="text-center pt-3">
                                    <MultiStateToggleButton
                                        options={[{ name: 'Month', value: '1' }, { name: 'Week', value: '2' }, { name: 'Lecture', value: '3' }]}
                                        currentActive={this.state.currentBookingFilter}
                                        handleClick={this.handleBookingFilterChange}
                                    />
                                </div>
                            </div>
                            <div className='col-8 mx-auto my-5 shadow-lg p-3 bg-white'>
                                {this.state.cancellationSeries != undefined &&
                                    <ApexChart options={this.state.cancellationOptions} series={this.state.cancellationSeries} type="line" height={350} />}
                                <div className="text-center pt-3">
                                    <MultiStateToggleButton
                                        options={[{ name: 'Month', value: '4' }, { name: 'Week', value: '5' }, { name: 'Lecture', value: '6' }]}
                                        currentActive={this.state.currentCancellationFilter}
                                        handleClick={this.handleCancellationFilterChange}
                                    />
                                </div>
                            </div>
                        </>
                        : /* no course selected: aggregate (generic) stats */
                        <>
                            <div className='col-8 mx-auto my-5 shadow-lg p-3 bg-white'>
                                {this.state.bookingSeries != undefined &&
                                    <ApexChart options={this.state.bookingOptions} series={this.state.bookingSeries} type="area" height={350} />}
                                <div className="text-center pt-3">
                                    <MultiStateToggleButton
                                        options={[{ name: 'Month', value: '1' }, { name: 'Week', value: '2' }]}
                                        currentActive={this.state.currentBookingFilter}
                                        handleClick={this.handleBookingFilterChange}
                                    />
                                </div>
                            </div>
                            <div className='col-8 mx-auto my-5 shadow-lg p-3 bg-white'>
                                {this.state.cancellationSeries != undefined &&
                                    <ApexChart options={this.state.cancellationOptions} series={this.state.cancellationSeries} type="line" height={350} />}
                                <div className="text-center pt-3">
                                    <MultiStateToggleButton
                                        options={[{ name: 'Month', value: '4' }, { name: 'Week', value: '5' }]}
                                        currentActive={this.state.currentCancellationFilter}
                                        handleClick={this.handleCancellationFilterChange}
                                    />
                                </div>
                            </div>
                        </>
                }

            </Container>
        </>
    }

    handleBookingFilterChange = (event) => {
        if (event === 'Month') {
            this.setState({ currentBookingFilter: 1 });
            API.getBookingStatisticsByMonth(this.state.selectedCourse).then((stats) => {
                let data = []; let labels = [];
                stats.forEach(stat => {
                    data.push(stat.average);
                    labels.push(`${stat.monthOfYear}-${stat.year}`);
                });
                this.updateBookingGraphData(
                    'Average number of booking per month',
                    data,
                    data, // tmp (attendance)
                    labels
                )
            });
        }
        else if (event === 'Week') {
            this.setState({ currentBookingFilter: 2 });
            API.getBookingStatisticsByWeek(this.state.selectedCourse).then((stats) => {
                let data = []; let labels = [];
                stats.forEach(stat => {
                    data.push(stat.average);
                    labels.push(`${stat.weekOfYear} (${stat.monthOfYear}-${stat.year})`);
                });
                this.updateBookingGraphData(
                    'Average number of booking per week',
                    data,
                    data, // tmp (attendance)
                    labels
                )
            });
        }
        else if (event === 'Lecture') {
            this.setState({ currentBookingFilter: 3 });
            API.getBookingStatisticsByLesson(this.state.selectedCourse).then((stats) => {
                let data = []; let labels = [];
                stats.forEach(stat => {
                    data.push(stat.numberBookings);
                    labels.push(`${stat.lectureID} - ${stat.dateLecture}`);
                });
                this.updateBookingGraphData(
                    'Number of booking per lecture',
                    data,
                    data, // tmp (attendance)
                    labels
                )
            });
        }
    }

    handleCancellationFilterChange = (event) => {
        if (event === 'Month') {
            this.setState({ currentCancellationFilter: 4 });
            this.updateCancellationGraphData(
                'Average number of cancellations per month',
                [40.5, 42.8, 45.7, 60.1, 50.2, 40.3, 30.2],
                ['Oct 2020', 'Nov 2020', 'Dec 2020', 'Jan 2021', 'Feb 2021', 'Mar 2021', 'Apr 2021']
            );
        }
        else if (event === 'Week') {
            this.setState({ currentCancellationFilter: 5 });
            this.updateCancellationGraphData(
                'Average number of cancellations per week',
                [40.5, 32.8, 28.7, 25.1],
                ['2/11 - 6/11', '9/11 - 13/11', '16/11 - 20/11', '23/11 - 27/11']
            );
        }
        else if (event === 'Lecture') {
            this.setState({ currentCancellationFilter: 6 });
            this.updateCancellationGraphData(
                'Number of cancellations per lecture',
                [5, 2, 0, 1, 10, 3, 2],
                ['SE II - 24/11 14:30', 'SE II - 25/11 14:30', 'SE II - 26/11 11:30', 'SE II - 26/11 14:30', 'SE II - 27/11 14:30', 'SE II - 28/11 14:30', 'SE II - 29/11 14:30']
            );
        }
    }

    handleSelectionChange = (selection) => {
        this.setState(
            {
                selectedCourse: selection !== null ? selection.idCourse : API.ALL_COURSES_FILTER,
                currentBookingFilter: 1,
                currentCancellationFilter: 4,
            },
            function() {
                // after state update
                API.getBookingStatisticsByMonth(this.state.selectedCourse).then((stats) => {
                    let data = []; let labels = [];
                    stats.forEach(stat => {
                        data.push(stat.average);
                        labels.push(`${stat.monthOfYear}-${stat.year}`);
                    });
                    this.updateBookingGraphData(
                        'Average number of booking per month',
                        data,
                        data, // tmp (attendance)
                        labels
                    )
                });
            }
        )

        this.updateCancellationGraphData(
            'Average number of cancellations per month',
            [40.5, 42.8, 45.7, 60.1, 50.2, 40.3, 30.2],
            ['Oct 2020', 'Nov 2020', 'Dec 2020', 'Jan 2021', 'Feb 2021', 'Mar 2021', 'Apr 2021']
        );
    }

    updateBookingGraphData = (title, yBookingData, yAttendanceData, xlabel) => {
        this.setState({
            bookingSeries: [
                {
                    name: 'Booking',
                    data: yBookingData
                },
                {
                    name: 'Attendance',
                    data: yAttendanceData
                }
            ],
            bookingOptions: {
                chart: {
                    height: 350,
                    type: 'area',
                    zoom: {
                        enabled: true
                    }
                },
                dataLabels: {
                    enabled: false
                },
                stroke: {
                    curve: 'smooth'
                },
                title: {
                    text: title,
                    align: 'center'
                },
                grid: {
                    row: {
                        colors: ['#f3f3f3', 'transparent'], // takes an array which will be repeated on columns
                        opacity: 0.5
                    },
                },
                xaxis: {
                    categories: xlabel
                },
                yaxis: {
                    labels: {
                        formatter: function (value) {
                            return value.toFixed(2);
                        }
                    },
                },
            },
        });
    }

    updateCancellationGraphData = (title, ydata, xlabel) => {
        this.setState({
            cancellationSeries: [{
                name: "Cancellation",
                data: ydata
            }],
            cancellationOptions: {
                chart: {
                    height: 350,
                    type: 'line',
                    zoom: {
                        enabled: true
                    }
                },
                dataLabels: {
                    enabled: false
                },
                stroke: {
                    curve: 'straight'
                },
                title: {
                    text: title,
                    align: 'center'
                },
                grid: {
                    row: {
                        colors: ['#f3f3f3', 'transparent'], // takes an array which will be repeated on columns
                        opacity: 0.5
                    },
                },
                xaxis: {
                    categories: xlabel,
                },
                yaxis: {
                    labels: {
                        formatter: function (value) {
                            return value.toFixed(2);
                        }
                    },
                },
            },
        });
    }
}

class MultiStateToggleButton extends React.Component {

    render() {
        return <>
            <ButtonGroup toggle>
                {this.props.options.map(o => this.createListItem(o))}
            </ButtonGroup>
        </>
    }

    createListItem = (option) => {

        return <ToggleButton
            key={option.value}
            type="radio"
            variant="dark"
            name="option"
            value={option.value}
            checked={this.props.currentActive == option.value}
            onClick={(event) => { event.preventDefault(); this.props.handleClick(event.currentTarget.innerText); }}
        >
            {option.name}
        </ToggleButton>
    }

}

export default BookingManagerPage;