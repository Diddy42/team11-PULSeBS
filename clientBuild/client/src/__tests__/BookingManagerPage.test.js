import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import BookingManagerPage from '../BookingManagerPage';
import React from 'react';
import API from '../API';
import { shallow } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('renders learn react link', () => {
    const tree = TestRenderer.create(<BookingManagerPage/>).toJSON();
    expect(tree).toMatchSnapshot();
  });