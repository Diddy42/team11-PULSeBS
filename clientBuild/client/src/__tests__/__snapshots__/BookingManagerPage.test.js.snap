// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="container-fluid"
>
  <br />
  <h1
    className="text-center"
  >
    Historical data
  </h1>
  <div
    className="col-4 mx-auto my-4 css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className=" css-yk16xz-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className=" css-g1d714-ValueContainer"
      >
        <div
          className=" css-1wa3eu0-placeholder"
        >
          All Courses
        </div>
        <div
          className="css-b8ldur-Input"
        >
          <div
            className=""
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-2-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className=" css-1hb7zxy-IndicatorsContainer"
      >
        <span
          className=" css-1okebmr-indicatorSeparator"
        />
        <div
          aria-hidden="true"
          className=" css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-6q0nyr-Svg"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="m-5 shadow-lg p-3 bg-white"
  >
    <div
      className="text-center pt-3"
    >
      <div
        className="btn-group btn-group-toggle"
        role="group"
      >
        <label
          className="btn btn-dark"
          disabled={false}
          onClick={[Function]}
        >
          <input
            autoComplete="off"
            checked={false}
            disabled={false}
            name="option"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value="1"
          />
          Month
        </label>
        <label
          className="btn btn-dark"
          disabled={false}
          onClick={[Function]}
        >
          <input
            autoComplete="off"
            checked={false}
            disabled={false}
            name="option"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value="2"
          />
          Week
        </label>
      </div>
    </div>
  </div>
  <div
    className="m-5 shadow-lg p-3 bg-white"
  >
    <div
      height={350}
      options={
        Object {
          "chart": Object {
            "height": 350,
            "type": "line",
            "zoom": Object {
              "enabled": true,
            },
          },
          "dataLabels": Object {
            "enabled": false,
          },
          "grid": Object {
            "row": Object {
              "colors": Array [
                "#f3f3f3",
                "transparent",
              ],
              "opacity": 0.5,
            },
          },
          "stroke": Object {
            "curve": "straight",
          },
          "title": Object {
            "align": "center",
            "text": "Average number of cancellations per month",
          },
          "xaxis": Object {
            "categories": Array [
              "Oct 2020",
              "Nov 2020",
              "Dec 2020",
              "Jan 2021",
              "Feb 2021",
              "Mar 2021",
              "Apr 2021",
            ],
            "convertedCatToNumeric": true,
          },
          "yaxis": Object {
            "labels": Object {
              "formatter": [Function],
            },
          },
        }
      }
      series={
        Array [
          Object {
            "data": Array [
              40.5,
              42.8,
              45.7,
              60.1,
              50.2,
              40.3,
              30.2,
            ],
            "name": "Cancellation",
          },
        ]
      }
      type="line"
      width="100%"
    />
    <div
      className="text-center pt-3"
    >
      <div
        className="btn-group btn-group-toggle"
        role="group"
      >
        <label
          className="btn btn-dark"
          disabled={false}
          onClick={[Function]}
        >
          <input
            autoComplete="off"
            checked={false}
            disabled={false}
            name="option"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value="4"
          />
          Month
        </label>
        <label
          className="btn btn-dark"
          disabled={false}
          onClick={[Function]}
        >
          <input
            autoComplete="off"
            checked={false}
            disabled={false}
            name="option"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value="5"
          />
          Week
        </label>
      </div>
    </div>
  </div>
</div>
`;
